spec/features/photos_feature_spec.rb

```diff
@@ -0,0 +1,121 @@
+      click_link "Add photo"
+      fill_in "Title", with: "Beach photo"
+      fill_in "Description", with: "Not actually a photo to do with the beach"
+      expect { click_button "Add Photo" }.not_to change(Photo, :count)
+      expect(page).to have_content "You must select a photo to upload"
+    end
+    scenario "title and description can be left blank" do
+      click_link "Add photo"
+      attach_file "photo_image", Rails.root.join("spec","fixtures", "test_photo.png")
+      expect { click_button "Add Photo" }.to change(Photo, :count).by(1)
+    end
+  end
+  context "photos can be viewed individually" do
+    scenario "by anyone" do
+      add_photo
+      click_link "Sign out"
+      photo = Photo.last
+      click_link "#{photo.id}"
+      expect(page).to have_xpath "//img[contains(@src,'test_photo.png')]"
+      expect(page).to have_content "Beach photo"
+      expect(page).to have_content "Not actually a photo to do with the beach"
+      expect(page).not_to have_link "Edit Photo"
+    end
+    context "by the photo owner" do
+      scenario "from their own page" do
+        add_photo
+        photo = Photo.last
+        click_link "#{ photo.id }"
+        expect(page).to have_xpath "//img[contains(@src,'test_photo.png')]"
+        expect(page).to have_content "Beach photo"
+        expect(page).to have_content "Not actually a photo to do with the beach"
+        expect(page).to have_link "Edit Photo"
+      end
+      scenario "from the home page" do
+        add_photo
+        click_link "Home"
+        photo = Photo.last
+        click_link "#{ photo.id }"
+        expect(page).to have_xpath "//img[contains(@src,'test_photo.png')]"
+        expect(page).to have_content "Beach photo"
+        expect(page).to have_content "Not actually a photo to do with the beach"
+        expect(page).to have_link "Edit Photo"
+      end
+    end
+    context "users can edit/delete photos" do
+      before {
+        add_photo
+        @photo = Photo.last
+        click_link "#{ @photo.id }"
+        click_link "Edit Photo"
+      }
+      scenario "users can edit their photo information" do
+        expect(page).to have_field("Title", with: "#{ @photo.title }")
+        expect(page).to have_field("Description", with: "#{ @photo.description }")
+        fill_in "Title", with: "Tetris score"
+        fill_in "Description", with: "Just a random high score from when I played tetris"
+        click_button "Edit Photo"
+        photo_refresh = Photo.find(@photo.id)
+        expect(photo_refresh.title).to eq "Tetris score"
+        expect(photo_refresh.description).to eq "Just a random high score from when I played tetris"
+        expect(current_path).to eq "/photos/#{ @photo.id }"
+      end
+      scenario "users can delete their photos" do
+        click_link "Delete Photo"
+        photo_refresh = 
+        expect { Photo.find(@photo.id) }.to raise_error "Couldn't find Photo with 'id'=#{ @photo.id }"
```

Comment:
Since you're dealing with a web interface, don't expect raised errors. Feature tests shouldn't care about how things are implemented, only that they _work_ at the interface level.